### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by osk**
* Turn of color theme for pretty-errors, I can't read the errors in my lightly-themed terminal because of low contrast
* Add ErrorBoundary component for client render
* Add catch() to SSR render

Should get rid of the pesky

```
SERVER: Error in server execution, check the console for more info.
(node:59606) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:59606) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

errors, which are most often triggered by some errors in the store but the errors get eaten and you're left going 🤔

Test by e.g. commenting out this line
https://github.com/ueno-llc/starter-kit-universally/blob/error-handling/shared/store/index.js#L8

### CODE REVIEW

**Review by hrafne — APPROVED** [review]

File: shared/components/error-boundary/ErrorBoundary.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    });
+  }
+
+  render() {
+    if (this.state.errorInfo) {
+      return (
+        <div>
+          <h2>Error</h2>
+          <p>{this.state.error && this.state.error.toString()}</p>
+          <pre>{this.state.errorInfo.componentStack}</pre>
```
Comment (birkir) [inline shared/components/error-boundary/ErrorBoundary.js:28]: What is `componentStack`?

**Comment by birkir** [comment]
Looks good, gotta love React 16!

File: shared/components/error-boundary/ErrorBoundary.js (hunk 1 above)
Comment (osk) [reply shared/components/error-boundary/ErrorBoundary.js:28]: It's contains the stack of components that caused the throw, see https://reactjs.org/docs/error-boundaries.html#componentdidcatch-parameters

**Comment by osk** [comment]
There's a pretty annoying problem with this and hot reloading. If an error is triggered via hot reload the error boundary is shown but on fixing the error it doesn't get hot reloaded away.

Updating to react-hot-reloader 4.0 will probably fix this, see:
https://github.com/gaearon/react-hot-loader#error-reporter-is-gone

It also should fix the decorator problem! No more temporary variables when exporting `inject`ed and `withJob`ed components! 😱
https://github.com/gaearon/react-hot-loader/issues/279#issuecomment-369409749

**Comment by birkir** [comment]
LGTM and then deal with HMR in another PR

**Pull request merged**